express-error-handling/server/index.js

```diff
@@ -0,0 +1,148 @@
+require('dotenv/config');
+const pg = require('pg');
+const express = require('express');
+const ClientError = require('./client-error'); // eslint-disable-line
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/grades', (req, res, next) => {
+  const { name, course } = req.body;
+  const score = parseInt(req.body.score, 10);
+  if (!name || !course || !score) {
+    throw new ClientError(400, `gradeId must be a positive integer`)
+    return;
```

Comment:
A throw statement will stop all current code execution and jump to the nearest `catch` so you don't need a `return` after you `throw` an error.